checks/shell_download_validate.go

```diff
@@ -552,57 +459,69 @@ func extractInterpreterCommandFromNode(node syntax.Node) (string, bool) {
 	return cs, true
 }
 
-func extractInterpreterCommandFromString(cmd string) (c string, res bool, err error) {
-	in := strings.NewReader(cmd)
+func nodeToString(p *syntax.Printer, node syntax.Node) (string, error) {
+	// https://github.com/mvdan/sh/blob/24dd9930bc1cfc7be025f8b75b2e9e9f04524012/syntax/printer.go#L135.
+	var buf bytes.Buffer
+	err := p.Print(&buf, node)
+	// This is ugly, but the parser does not have a defined error type :/.
+	if err != nil && !strings.Contains(err.Error(), "unsupported node type") {
```

Comment:
what about other non-nil errors? do we not care?

Reply:
this line will catch all errors that are *not* of the type `unsupported type`. We're only discarding the `unsupported type` because those won't need to be printed, ie they will never contain insecure download code.

Reply:
I see, I misunderstood the intent here. Would be good to add a comment here to explain why we discard "unsupported node type" :)

Reply:
will do. Thanks